var/ramble/repos/builtin/applications/intel-hpl/application.py

```diff
@@ -26,27 +26,66 @@ class IntelHpl(ExecutableApplication):
 
     tags("benchmark-app", "benchmark", "linpack", "optimized", "intel", "mkl")
 
-    define_compiler("gcc9", pkg_spec="gcc@9.3.0", package_manager="spack*")
+    define_compiler("gcc13p2", pkg_spec="gcc@13.2.0", package_manager="spack*")
     software_spec(
-        "imkl_2023p1",
-        pkg_spec="intel-oneapi-mkl@2023.1.0 threads=openmp",
-        compiler="gcc9",
+        "imkl_2024p2",
+        pkg_spec="intel-oneapi-mkl@2024.2.0 threads=openmp",
+        compiler="gcc13p2",
         package_manager="spack*",
     )
     software_spec(
-        "impi_2018", pkg_spec="intel-mpi@2018.4.274", package_manager="spack*"
+        "impi2021p11",
+        pkg_spec="intel-oneapi-mpi@2021.11.0",
+        package_manager="spack*",
     )
 
     required_package("intel-oneapi-mkl", package_manager="spack*")
 
+    # This step does a few things:
+    # - Set up the env variables expected by runme_intel64_prv
+    #   (We call this runner script instead of the underlying xhpl_intel64_dynamic
+    #    since it sets up derived env var HPL_HOST_NODE for numa placement control.)
+    # - Link in the xhpl_intel64_dynamic binary to the running dir
+    #   (This is needed due to runme_intel64_prv invoking it using "./")
+    # - Account for newer directory layout from mkl 2024
+    executable(
+        "prepare",
+        template=[
+            r"""
+export MPI_PROC_NUM="{n_ranks}"
+export MPI_PER_NODE="{processes_per_node}"
+export NUMA_PER_MPI="{numa_per_mpi}"
+export HPL_EXE=xhpl_intel64_dynamic
```

Comment:
For these environment variables (that are explicitly exported), do you think we could use this:
https://github.com/GoogleCloudPlatform/ramble/blob/develop/lib/ramble/ramble/language/application_language.py#L245

Instead if exporting them directly within the executable?

Reply:
Good call, updated.